### DISCUSSION TITLE
Allow accessing data of reference metafields

### DISCUSSION THREAD

**Discussion by bakura10** [Shopify/function-examples#332] [Feedback: General Improvements]
Hi,

I did some experiment with Shopify Function for a client. The use case would be to retrieve a customer metafield a functions. This metafield is a reference to a metaobject.

While I could successfully retrieve the metafield, I only got the metaobject ID, from which I can't do anything. It would be useful if we could use the `reference` connection to retrieve the associated object (metaobject would be useful of course, but any other reference ; product, collection...) would be also useful.

Thanks

**Comment by nboliver-ventureweb** [comment]
Agreed, this would be a very useful feature. Passing bundle configurations through as JSON metafields is a lot more work / clunky than  passing a single metafield with a metaobject that references different variants.

**Comment by KthProg** [comment]
This would be very helpful

**Comment by nboliver-ventureweb** [comment]
Cart metafields were just released. There may be a way to leverage that to pass in dynamic configuration data, but still a lot more complex than just accessing a metaobject.
